src/core_ocean/shared/mpas_ocn_tidal_potential_forcing.F

```diff
@@ -0,0 +1,663 @@
+        call mpas_pool_get_array(forcingPool, 'tidalPotentialConstituentType', tidalConstituentType)
+        call mpas_pool_get_array(forcingPool, 'tidalPotentialLatitudeFunction', latitudeFunction)
+        call mpas_pool_get_array(meshPool, 'latCell', latCell)
+        call mpas_pool_get_dimension(meshPool, 'nCells', nCells)
+
+        call mpas_set_time(refTime, dateTimeString=config_tidal_potential_reference_time)
+
+        nTidalConstituents = 0
+        if (config_use_tidal_potential_forcing_M2) then
+          nTidalConstituents = nTidalConstituents + 1
+          constituentList(nTidalConstituents)%constituent = 'M2'
+        end if 
+
+        if (config_use_tidal_potential_forcing_S2) then
+          nTidalConstituents = nTidalConstituents + 1
+          constituentList(nTidalConstituents)%constituent = 'S2'
+        end if 
+ 
+        if (config_use_tidal_potential_forcing_N2) then
+          nTidalConstituents = nTidalConstituents + 1
+          constituentList(nTidalConstituents)%constituent = 'N2'
+        end if 
+
+        if (config_use_tidal_potential_forcing_K2) then
+          nTidalConstituents = nTidalConstituents + 1
+          constituentList(nTidalConstituents)%constituent = 'K2'
+        end if 
+
+        if (config_use_tidal_potential_forcing_K1) then
+          nTidalConstituents = nTidalConstituents + 1
+          constituentList(nTidalConstituents)%constituent = 'K1'
+        end if 
+
+        if (config_use_tidal_potential_forcing_O1) then
+          nTidalConstituents = nTidalConstituents + 1
+          constituentList(nTidalConstituents)%constituent = 'O1'
+        end if 
+
+        if (config_use_tidal_potential_forcing_Q1) then
+          nTidalConstituents = nTidalConstituents + 1
+          constituentList(nTidalConstituents)%constituent = 'Q1'
+        end if 
+
+        if (config_use_tidal_potential_forcing_P1) then
+          nTidalConstituents = nTidalConstituents + 1
+          constituentList(nTidalConstituents)%constituent = 'P1'
+        end if 
+
+        call tidal_constituent_factors(constituentList,nTidalConstituents,refTime, &
+                                       tidalConstituentFreqency, &
+                                       tidalConstituentAmplitude, &
+                                       tidalConstituentLoveNumbers, &
+                                       tidalConstituentNodalAmplitude, &
+                                       tidalConstituentAstronomical, &
+                                       tidalConstituentNodalPhase, &
+                                       tidalConstituentType, &
+                                       err)
+
+        do iCell = 1,nCells
+          lat = latCell(iCell)
+          latitudeFunction(iCell,1) = 3.0_RKIND*sin(lat)**2 - 1.0_RKIND 
+          latitudeFunction(iCell,2) = sin(2.0_RKIND*lat)
+          latitudeFunction(iCell,3) = cos(lat)**2
+        end do
+        
+        do iCon = 1,nTidalConstituents  
```

Comment:
Nice, I think this will be helpful for debugging and documentation and since it is on init having this level of output should be ok.

Reply:
Thanks, I think it's worth keeping in there.